src/set_theory/ordinal_arithmetic.lean

```diff
@@ -1021,58 +1021,66 @@ rfl
 
 /-- The supremum of a family of ordinals -/
 def sup {ι : Type u} (f : ι → ordinal.{max u v}) : ordinal.{max u v} :=
-Inf {c | ∀ i, f i ≤ c}
+Sup (set.range f)
```

Comment:
Can't you get rid of the definition altogether? Up to some universes, this is just `⨆ i, f i` now.

Reply:
Should I? There's a slight inconvenience in doing so, which is that you now have to invoke the proof that the supremum exists, instead of being inferred from the universes. You also lose a bunch of symmetry in the API, but that's less important.

Reply:
How can replacing a declaration by its definition break anything? You can still type-ascript `ι` and `f` as you do now.

Reply:
You're right, I'll get to work on this.

Reply:
Turns out this will need quite a large refactor. The reason is, changing `sup` by `supr` allows almost all of the theorems to be made fully universe-polymorphic, given a proof of `bdd_above`. Perhaps I could do this in a follow-up PR? This already changes a lot of things as is.

Reply:
Also, using `supr` instead of `sup` without the `bdd_above` condition means that many, many universe ascriptions need to be added all over the place. So I'd rather keep this as is for now.